src/commands/Register.ts

```diff
@@ -18,130 +18,207 @@ export type RegisterOptions = {
-    } else {
-      log.info('Funding relayer');
+  if (unstakeDelay.gte(options.unstakeDelay) && stake.gte(options.stake)) {
+    log.info('Relayer already staked');
+  } else {
+    const stakeValue = options.stake.sub(stake);
+    log.info(
+      `Staking relayer ${utils.formatUnits(stakeValue, 'ether')} RBTC`,
+      stake.eq(constants.Zero)
+        ? ''
+        : ` (already has ${utils.formatUnits(stake, 'ether')} RBTC)`
+    );
 
-      const fundTx = await options.signer.sendTransaction({
-        to: relayAddress,
-        value: options.funds,
+    const stakeTx = await relayHub
+      .connect(options.signer)
+      .stakeForAddress(relayAddress, options.unstakeDelay.toString(), {
+        value: stakeValue,
         gasLimit: 1e6,
         gasPrice: options.gasPrice,
       });
 
-      if (fundTx.hash == null) {
-        throw new Error(`Fund transaction reverted: ${JSON.stringify(fundTx)}`);
-      }
-      transactions.push(fundTx.hash);
-    }
+    transactions.push(stakeTx.hash);
+  }
 
-    await this.waitForRelay(options.relayUrl);
-    log.info('Executed Transactions', transactions);
+  if (isSameAddress(owner, from)) {
+    log.info('Relayer already authorized');
   }
-}
 
-export async function executeRegister(registerOptions?: RegisterOptions) {
-  const { app, contracts, blockchain } = getServerConfig();
-  let registerConfig: RegisterConfig | undefined = undefined;
-  if (config.has('register')) {
-    registerConfig = config.get('register');
+  const bal = await provider.getBalance(relayAddress);
+
+  if (bal.gt(options.funds)) {
+    log.info('Relayer already funded');
+  } else {
+    log.info(
+      `Funding relayer ${utils.formatUnits(options.funds, 'ether')} RBTC`
+    );
+
+    const fundTx = await options.signer.sendTransaction({
+      to: relayAddress,
+      value: options.funds,
+      gasLimit: 1e6,
+      gasPrice: options.gasPrice,
+    });
+
+    if (fundTx.hash == null) {
+      throw new Error(`Fund transaction reverted: ${JSON.stringify(fundTx)}`);
+    }
+    transactions.push(fundTx.hash);
   }
+
+  await waitForRelay(httpClient, options.relayUrl);
+  log.info('Executed Transactions', transactions);
+};
+
+export async function executeRegister() {
+  // FIXME: add registerOptions?: RegisterOptions from either config or cli (which one?)
+  const { app, contracts, blockchain } = getServerConfig();
   log.setLevel(app.logLevel);
-  const register = new Register(
-    blockchain.rskNodeUrl,
-    registerConfig?.mnemonic
-  );
+  log.debug('configSources', config.util.getConfigSources());
+  const { account, stake, funds, mnemonic }: RegisterConfig = config.has(
+    'register'
+  )
+    ? config.get('register')
+    : {};
+
+  if (account && !mnemonic) {
+    log.error(`
+    You must configure mnemonic for given account address.
+    `);
+  }
+
+  const rpcProvider = new JsonRpcProvider(blockchain.rskNodeUrl);
   const portIncluded: boolean = app.url.indexOf(':') > 0;
```

Comment:
```suggestion
  const portIncluded = !!app.url.match(/:\d+$/);
```

Sorry I think that we always had this wrong and was working because we were always including the port in the address.

Reply:
changed it now